transport/server.cc

```diff
@@ -1078,7 +1079,13 @@ process_query_internal(service::client_state& client_state, distributed<cql3::qu
             return cql_server::process_fn_return_type(convert_error_message_to_coordinator_result(msg.get()));
         } else {
             tracing::trace(q_state->query_state.get_trace_state(), "Done processing - preparing a result");
-            return cql_server::process_fn_return_type(make_foreign(make_result(stream, *msg, q_state->query_state.get_trace_state(), version, skip_metadata)));
+
+            std::optional<seastar::lw_shared_ptr<cql3::cql_metadata_id_type>> metadata_id_opt;
```

Comment:
Why lw_shared_ptr? It adds an extra allocation.

Reply:
Unless it is typically copied rather than newly materialized. I think that's the case.